baker_cli.go

```diff
@@ -0,0 +1,181 @@
+package baker
+
+import (
+	"flag"
+	"fmt"
+	"html/template"
+	"math/rand"
+	"net"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	flagHelpConfig = flag.String("help", "", "show help for a `component` (input/filter/output/upload) (use '*' to dump all)")
+	flagVersion    = flag.Bool("version", false, "print build version number")
+	flagVerbose    = flag.Bool("v", false, "verbose logging (debug level)")
+	flagQuiet      = flag.Bool("q", false, "quiet logging (warn level)")
+	flagPretty     = flag.Bool("pretty", false, "human-readable logging (unstructured logging)")
+	flagPProf      = flag.String("pprof", "", `run pprof server on host port provided (disabled if ""), use "localhost:"  for a free port`)
+)
+
+// Use `-ldflags="-X 'github.com/AdRoll/baker.BuildVersion=someversion'"` when building Baker to set this value
+var BuildVersion = "-- unknown --"
+
+// MainCLI provides a handy way to quickly create a Baker program only providing
+// the available Components.
+//
+// The function includes many utilities that can be configured by command line arguments:
+// -help: Prints available options and components
+// -version: print build version (build with `-ldflags="-X 'github.com/AdRoll/baker.BuildVersion=someversion'"` to set the value)
+// -v: verbose logging (not compatible with -q)
+// -q: quiet logging (not compatible with -v)
+// -pretty: logs in textual format instead of JSON format
+// -pprof: run a pprof server on the provided host:port address
```

Comment:
In generated go doc, `// hello` and `//  hello` (2 spaces) are formatted differently.
With 2 spaces between `//` and the text, the text shows as code block in godoc. Using a code block for the list of cli arguments here could maybe look better